### PR TITLE
Created endpoint to load trajectories from id and a date

### PR DESCRIPTION
**Pull request by Flaiizor**
Here you go Kristoffer

### CODE REVIEW

File: src/boost-api/Endpoint_Handlers.cpp [hunk 1]
```diff
@@ -46,6 +46,27 @@ namespace api {
         return json_data;
     }
 
+    boost::json::array get_json_trajectory(const std::vector<data_structures::Trajectory>& trajectory){
```
Comment (Kristoffermg) [inline src/boost-api/Endpoint_Handlers.cpp]: Kunne måske omnavngives til "convert_trajectory_to_json"

**Review by Kristoffermg — CHANGES_REQUESTED** [review]
Jeg tror det bliver godt

File: src/boost-api/Endpoint_Handlers.cpp (hunk 1 above)
Comment (Flaiizor) [reply src/boost-api/Endpoint_Handlers.cpp]: Vi har omnavngivet nu, venligst afslut denne samtale

**Review by Kristoffermg — APPROVED** [review]
worst case) XML, right?

For your white box / extra space issue, try

style="display: none;"
instead of

style="visibility: hidden;"
[Share](https://stackoverflow.com/a/4084700)
[Improve this answer](https://stackoverflow.com/posts/4084700/edit)
Follow
answered Nov 3, 2010 at 6:34
[coolaj86](https://stackoverflow.com/users/151312/coolaj86)'s user avatar
coolaj86
75.5k2121 gold badges108108 silver badges129129 bronze badges
yes actually I do, well I dont WANT one but it is required by the client – 
[mattgcon](https://stackoverflow.com/users/173923/mattgcon)
 [Nov 3, 2010 at 6:41](https://stackoverflow.com/questions/4084674/make-one-div-visible-and-another-invisible#comment4392534_4084700)
[Add a comment](https://stackoverflow.com/questions/4084674/make-one-div-visible-and-another-invisible#)
0

If u want to use display=block it will make the content reader jump, so instead of using display you can set the left attribute to a negative value which does not exist in your html page to be displayed but actually it do.

I hope you must be understanding my point, if I am unable to make u understand u can message me back.

[Share](https://stackoverflow.com/a/4084962)
[Improve this answer](https://stackoverflow.com/posts/4084962/edit)
Follow
answered Nov 3, 2010 at 7:31
[Prateek](https://stackoverflow.com/users/466782/prateek)'s user avatar
Prateek
1,5581313 silver badges2323 bronze badges
[Add a comment](https://stackoverflow.com/questions/4084674/make-one-div-visible-and-another-invisible#)
Your Answer
Sign up or [log in](https://stackoverflow.com/users/login?ssrc=question_page&returnurl=https%3a%2f%2fstackoverflow.com%2fquestions%2f4084674%2fmake-one-div-visible-and-another-invisible%23new-answer)
Post as a guest
Name
Email
Required, but never shown

By clicking “Post Your Answer”, you agree to our terms of service and acknowledge you have read our privacy policy.

Not the answer you're looking for? Browse other questions tagged csshtmlvisibilityhidden or ask your own question.
The Overflow Blog
What language should beginning programmers choose?
Supporting the world’s most-used database engine through 2050
Featured on Meta
New Focus Styles & Updated Styling for Button Groups
Upcoming initiatives on Stack Overflow and across the Stack Exchange network
Pausing the 1-rep voting experiment on Stack Overflow: reflecting on the...
Temporary policy: Generative AI (e.g., ChatGPT) is banned
Linked
-3
How to hide a button after clicking on it,at that time how to make a div visible?
Related
4
How can I controle the [visibility](https://stackoverflow.com/questions/tagged/visibility) level of a div and make it see through?
0
Using visibility: [hidden](https://stackoverflow.com/questions/tagged/hidden) and display: none together in CSS?
9
CSS display:none and visibility:hidden
1[4](https://stackoverflow.com/q/2138107?rq=3)2
[How to make a DIV visible and invisible with JavaScript?](https://stackoverflow.com/questions/9456289/how-to-make-a-div-visible-and-invisible-with-javascript?rq=3)
[18](https://stackoverflow.com/q/10392987?rq=3)
[visibility:visible/hidden div](https://stackoverflow.com/questions/10392987/visibilityvisible-hidden-div?rq=3)
[1](https://stackoverflow.com/q/19164738?rq=3)
[Toggle visibility of two divs at once](https://stackoverflow.com/questions/19164738/toggle-visibility-of-two-divs-at-once?rq=3)
[2](https://stackoverflow.com/q/20334808?rq=3)
[One div visible and other hidden](https://stackoverflow.com/questions/20334808/one-div-visible-and-other-hidden?rq=3)
[1](https://stackoverflow.com/q/23474171?rq=3)
[Changing visibility of the div element using only CSS](https://stackoverflow.com/questions/23474171/changing-visibility-of-the-div-element-using-only-css?rq=3)
[1](https://stackoverflow.com/q/28950611?rq=3)
[How to hide content one and show content two?](https://stackoverflow.com/questions/28950611/how-to-hide-content-one-and-show-content-two?rq=3)
[1](https://stackoverflow.com/q/40477235?rq=3)
[Hide one div and show another?](https://stackoverflow.com/questions/40477235/hide-one-div-and-show-another?rq=3)
[Hot Network Questions](https://stackexchange.com/questions?tab=hot)
[Is std::atomic<bool> trivially copyable?](https://stackoverflow.com/questions/78384904/is-stdatomicbool-trivially-copyable)
[Advantage of GLMs over transformation models](https://stats.stackexchange.com/questions/645797/advantage-of-glms-over-transformation-models)
[Do "tinker" and "tinkerer" imply "unskillful"?](https://ell.stackexchange.com/questions/351386/do-tinker-and-tinkerer-imply-unskillful)
[Why does no historian speak of the torn veil and how do we reconcile it with scripture?](https://hermeneutics.stackexchange.com/questions/94117/why-does-no-historian-speak-of-the-torn-veil-and-how-do-we-reconcile-it-with-scr)
[Who do I hire to help plan a significant project?](https://diy.stackexchange.com/questions/298719/who-do-i-hire-to-help-plan-a-significant-project)
[Evaluating a rational function integral in a quick way](https://math.stackexchange.com/questions/4905326/evaluating-a-rational-function-integral-in-a-quick-way)
[Why use a special "Name" class (instead of just a string) for representing objec](https://softwareengineering.stackexchange.com/questions/453043/why-use-a-special-name-class-instead-of-just-a-string-for-representing-objec)

**Review by Mailaur — APPROVED** [review]

File: src/boost-api/Endpoint_Handlers.cpp [hunk 2]
```diff
@@ -46,6 +46,27 @@ namespace api {
...
+            json_location["timestamp"] = loc.timestamp;
+            json_location["longitude"] = loc.longitude;
+            json_location["latitude"] = loc.latitude;
+            json_locations.push_back(std::move(json_location));
+        }
+
+        json_traj["locations"] = std::move(json_locations);
+        response_trajectory.push_back(std::move(json_traj));
+
+        return response_trajectory;
```
Comment (KarmaKamikaze) [inline src/boost-api/Endpoint_Handlers.cpp]: Return it `boost::json::object` i stedet for et array, hvis det er muligt.

File: src/trajectory_data_handling/Trajectory_Manager.cpp [hunk 3]
```diff
@@ -202,7 +203,59 @@ namespace trajectory_data_handling {
...
+              << " AND time >= '" << date_start << "' AND time <= '" << date_end << "';";
+
+        pqxx::result query_res{txn.exec(query.str())};
+        txn.commit();
+        std::vector<data_structures::Trajectory> res{};
+
+        data_structures::Trajectory working_trajectory{};
+        auto current_trajectory_id = query_res.front()["trajectory_id"];
+        if (current_trajectory_id.is_null())
+            throw std::runtime_error("Trajectory ID is null or not found in the database. Consider running simplification first!");
```
Comment (KarmaKamikaze) [inline src/trajectory_data_handling/Trajectory_Manager.cpp:235]: Det her crasher backend APIen, så sådan en fejl skal nok hånderes i en endpoint handler, således sådan et kald til denne funktion aldrig vil kunne finde sted.

**Review by KarmaKamikaze — CHANGES_REQUESTED** [review]

File: src/boost-api/Endpoint_Handlers.cpp (hunk 2 above)
Comment (Schwartz123) [reply src/boost-api/Endpoint_Handlers.cpp]: done

File: src/trajectory_data_handling/Trajectory_Manager.cpp (hunk 3 above)
Comment (Schwartz123) [reply src/trajectory_data_handling/Trajectory_Manager.cpp:235]: no

Comment (Flaiizor) [reply src/trajectory_data_handling/Trajectory_Manager.cpp:235]: Desværre crasher det ikke. vi bruger kun denne funktion i en try catch i endpoint handler, så vi får altid bare en 400 bad request :)

**Review by KarmaKamikaze — APPROVED** [review]
lgtm

**Pull request merged**